lib/query.js

```diff
@@ -3004,20 +3004,25 @@ Query.prototype.findOneAndUpdate = function(criteria, doc, options, callback) {
     this._mergeUpdate(doc);
   }
 
-  if (options) {
-    options = utils.clone(options);
-    if (options.projection) {
-      this.select(options.projection);
-      delete options.projection;
-    }
-    if (options.fields) {
-      this.select(options.fields);
-      delete options.fields;
-    }
+  options = options ? utils.clone(options) : {};
 
-    this.setOptions(options);
+  if (options.projection) {
+    this.select(options.projection);
+    delete options.projection;
+  }
+  if (options.fields) {
+    this.select(options.fields);
+    delete options.fields;
+  }
+
+
+  const returnOriginal = get(this, 'model.base.options.returnOriginal');
+  if (options.new == null && returnOriginal != null) {
+    options.new = !returnOriginal;
```

Comment:
I'd recommend setting `options.returnOriginal = returnOriginal` for consistency's sake. [We convert `new` to `returnOriginal` under the hood anyway](https://github.com/Automattic/mongoose/blob/f168f17526fc2e468313ca2d857e5452f6c05031/lib/query.js#L3404-L3407).

Reply:
Sounds reasonable, I'll take a look at those changes in the next weekend, we can move this to 5.10.1 if it will delay things.